src/main/menu.js

```diff
@@ -0,0 +1,452 @@
+  */
+gpii.app.menu.getKeyedInSnapset = function (keyedInUserToken, snapsetName, keyedInStrTemp) {
+    var keyedInUser = null;
+
+    if (keyedInUserToken) {
+        keyedInUser = {
+            label: fluid.stringTemplate(keyedInStrTemp, {"snapsetName": snapsetName}),
+            enabled: false
+        };
+    }
+
+    return keyedInUser;
+};
+
+/**
+  * Generates an object that represents the menu item for keying out.
+  * @param keyedInUserToken {String} The user token that is currently keyed in.
+  * @param keyOutStr {String} The string to be displayed for the key out menu item
+  * if there is a keyed in user.
+  * @param notKeyedInStr {String} The string to be displayed when a user is not
+  * keyed in.
+  * @return {ElectronMenuItem}
+  */
+gpii.app.menu.getKeyOut = function (keyedInUserToken, keyOutStr, notKeyedInStr) {
+    var keyOut;
+
+    if (keyedInUserToken) {
+        keyOut = {
+            label: keyOutStr,
+            click: "onKeyOut",
+            args: {
+                token: keyedInUserToken
+            },
+            enabled: true
+        };
+    } else {
+        keyOut = {
+            label: notKeyedInStr,
+            enabled: false
+        };
+    }
+
+    return keyOut;
+};
+
+/**
+ * Generates an array that represents the menu items related to a
+ * user's preference sets. The returned array can be used in the
+ * context menu of a {Tray} object.
+ * @param preferenceSets {Array} An array of all preference sets for the user.
+ * @param activeSet {String} The path of the currently active preference set.
+ * @return {ElectronMenuItem[]}
+ */
+gpii.app.menu.getPreferenceSetsMenuItems = function (preferenceSets, activeSet) {
+    var preferenceSetsLabels,
+        separator = {type: "separator"};
+
+    preferenceSetsLabels = preferenceSets.map(function (preferenceSet) {
+        return {
+            label: preferenceSet.name,
+            type: "radio",
+            args: {
+                path: preferenceSet.path
+            },
+            click: "onActivePrefSetUpdate",
+            checked: preferenceSet.path === activeSet
+        };
+    });
+
+    if (preferenceSetsLabels.length > 0) {
+        preferenceSetsLabels.unshift(separator);
+        preferenceSetsLabels.push(separator);
+    }
+
+    return preferenceSetsLabels;
+};
+
+/**
+  * Generates an object that represents the menu items for opening the settings panel
+  * @param keyedInUserToken {String} The user token that is currently keyed in.
+  * @param openSettingsStr {String} The string to be displayed for the open setting panel menu item.
+  * @returns {ElectronMenuItem}
+  */
+gpii.app.menu.getShowPSP = function (keyedInUserToken, openSettingsStr) {
+    return {
+        label: openSettingsStr,
+        click: "onPSP",
+        args: {
+            token: keyedInUserToken
+        }
+    };
+};
+
+/**
+  * Creates a JSON representation of a menu.
+  * @param {Object} An object containing a menu item template.
```

Comment:
Looks like this comment is not quite right, as the function seems to accept a set of "splatted" arguments